salt/utils/zeromq.py

```diff
@@ -5,21 +5,56 @@
 # Import Python libs
 from __future__ import absolute_import, print_function, unicode_literals
 
-# Import Salt libs
+import logging
+import tornado.ioloop
 from salt.exceptions import SaltSystemExit
 
-# Import 3rd-party libs
+log = logging.getLogger(__name__)
+
 try:
     import zmq
-    HAS_ZMQ = True
 except ImportError:
-    HAS_ZMQ = False
+    zmq = None
+    log.debug('ZMQ module is not found')
+
+ZMQDefaultLoop = None
+ZMQ_VERSION_INFO = (-1, -1, -1)
+LIBZMQ_VERSION_INFO = (-1, -1, -1)
+
+try:
+    if zmq:
+        ZMQ_VERSION_INFO = tuple([int(v_el) for v_el in zmq.__version__.split('.')])
+        LIBZMQ_VERSION_INFO = tuple([int(v_el) for v_el in zmq.zmq_version().split('.')])
+        if ZMQ_VERSION_INFO[0] > 16:  # 17.0.x+ deprecates zmq's ioloops
+            ZMQDefaultLoop = tornado.ioloop.IOLoop
+except Exception as ex:
```

Comment:
The `as ex` is still unnecessary here, since it won't be used. You don't need to assign the exception to a variable for log.exception to have access to the traceback. A simple `except Exception:` here is all that is needed.

Reply:
Fixd.